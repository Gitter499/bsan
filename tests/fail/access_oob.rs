//@run: 1
fn main() {
    let mut x = 1;
    let rx = &mut x;
    let ptr_x = rx as *mut i32;
    foo(ptr_x);
}

fn foo(p: *mut i32) {
    unsafe { p.add(1).write(0) }
}