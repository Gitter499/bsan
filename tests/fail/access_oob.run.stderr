An error occurred: UndefinedBehavior(AccessOutOfBounds(Provenance { alloc_id: ALLOC, bor_tag: <TAG>, alloc_info: $HEX }, 4, 4))


